### PR TITLE
Functionality Increment: Reallocate Command

### PR DESCRIPTION
**Pull request by jetnew**
This PR adds the functionality increment of the Reallocate Command. Reallocation of a student from one room to another room is performed by referencing the index of the student and the **new** room in the Address Book. `si/` and `ri/` prefixes refer to the student and new room indices respectively.

Example usage:
`reallocate si/1 ri/2`

Changes made:
* Add `ReallocateCommand`, that performs 3 checks on execution:
  * Student referenced exists.
  * New room referenced exists.
  * Student has previously been allocated a room.
* Add `ReallocateCommandParser`, that parses student and room indices from `si/` and `ri/ respectively.

To be done:
* Add testing support for `ReallocateCommand` and `ReallocateCommandParser`.

As part of #18.

### CODE REVIEW

**Comment by jetnew** [comment]
**User-test basic GUI and Storage functionality.** The purpose of these screenshots is to show basic user testing prior to requesting review.

GUI - Before Reallocate


GUI - After Reallocate


Storage - Before Reallocate


Storage - After Reallocate

File: src/main/java/seedu/resireg/logic/parser/ReallocateCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import seedu.resireg.logic.parser.exceptions.ParseException;
+
+
+/**
+ * Parses input arguments and creates a new AddCommand object
+ */
+public class ReallocateCommandParser implements Parser<ReallocateCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
```
Comment (mkeoliya) [inline src/main/java/seedu/resireg/logic/parser/ReallocateCommandParser.java]: `AddCommand` should be replace by `ReallocateCommand`

Comment (jetnew) [reply src/main/java/seedu/resireg/logic/parser/ReallocateCommandParser.java]: Fixed.

**Review by Lysire — APPROVED** [review]
LGTM!

**Pull request merged**